src/app/session/mock-session.service.ts

```diff
@@ -0,0 +1,73 @@
+import { SessionService } from './session.service';
+import { User } from 'app/user/user';
+import { StateHandler } from './util/state-handler';
+import { ConnectionState } from './connection-state.enum';
+import { LoginState } from './login-state.enum';
+import { SyncState } from './sync-state.enum';
+import { MockDatabase } from 'app/database/mock-database';
+import { Database } from 'app/database/database';
+import { EntityMapperService } from 'app/entity/entity-mapper.service';
+import { EntitySchemaService } from 'app/entity/schema/entity-schema.service';
+
+export class MockSessionService extends SessionService {
+    private database: MockDatabase;
+    private demoUser: User;
+    private loginState: StateHandler<LoginState> = new StateHandler<LoginState>(LoginState.LOGGED_OUT);
+    private connectionState: StateHandler<ConnectionState> = new StateHandler<ConnectionState>(ConnectionState.DISCONNECTED);
+    private syncState: StateHandler<SyncState> = new StateHandler<SyncState>(SyncState.UNSYNCED);
+
+    constructor(private _entitySchemaService: EntitySchemaService) {
+        super();
+        this.database = new MockDatabase();
+        this.createDemoUser();
+    }
+
+    private createDemoUser() {
+        const entityMapper = new EntityMapperService(this.database, this._entitySchemaService);
+
+        // add demo user
+        this.demoUser = new User('demo');
+        this.demoUser.name = 'demo';
+        this.demoUser.setNewPassword('pass');
+        entityMapper.save(this.demoUser);
+    }
+
+    public getCurrentUser(): User {
+        return this.demoUser;
+    }
+    public isLoggedIn(): boolean {
+        return this.loginState.getState() === LoginState.LOGGED_IN;
+    }
+    public getConnectionState(): StateHandler<ConnectionState> {
+        return this.connectionState;
+    }
+    public getLoginState(): StateHandler<LoginState> {
+        return this.loginState;
+    }
+    public getSyncState(): StateHandler<SyncState> {
+        return this.syncState;
+    }
+    public getDatabase(): Database {
+        return this.database;
+    }
+    public login(username, password): Promise<LoginState> {
+        if (username === 'demo' && password === 'pass') {

```

Comment:
It would be good if the login method uses the users of the mock BD. When working with the admin panel or creating new users it would be helpful if these things can also be tested in the temporary database.

Reply:
I'll have a look at this. As far as I remember, this piece was adapted from the old session/Database-Manager, but what you say makes sense.

Reply:
you could even consider some inheritance from LocalSession (`class MockSession extends LocalSession`) and override the diverting parts

Reply:
The `MockSession` must (at least indirectly) extend the abstract `Session` for Dependency Injection. I'd rather not make `LocalSession` extend `Session`, as we'd need to implement the additional APIs then that don't make sense at the `LocalSession`...

Reply:
@TheSlimvReal The demo user is now pulled from the database instead of being hardcoded. However, it is still defined in the MockSession – should we move it to the Demo Data generation?

Reply:
@sleidig This is my first time working with demo data generation, and I really like it! It feels much better to have the demo user in there (see https://github.com/Aam-Digital/ndb-core/pull/246/commits/9ede8b53a42de38a790641a9a55fd6ff2524454f)

Reply:
:-)